### ISSUE TITLE
Flutter Web - "flutter-loader" animation is not removed / stopped after main widget is loaded

### ISSUE THREAD

**Issue by Abhijit-Revamp** [flutter/flutter#85961]
## Steps to Reproduce

1. Create a flutter web project.
2. In the main.dart, set the background to be transparent or semi transparent
3. The horizontal loading animation (css class name "flutter-loader") continues to run
4. Looks very ODD with transparent / translucent backgrounds

**Expected results:**
Once the widget is loaded, the animation must stop or be removed

**Actual results:** 
The top edge continues to animate itself:
see attached image. My index.html has a top margin so the bar looks offset from top

![image](https://user-images.githubusercontent.com/44320018/124584078-74f7ce00-de71-11eb-8dfb-36f93f23383e.png)

**Comment by Abhijit-Revamp** [comment]
Workaround:
In main.dart main() function

```
  var loader = html.querySelector('.flutter-loader');
  loader?.className = '';
```

**Comment by darshankawar** [comment]
@Abhijit-Revamp 
Can you provide `flutter doctor -v` ?
I have seen the flutter-loader animation before, but didn't get it now.  I used below sample code:

```
return MaterialApp(
      home: Scaffold(
        backgroundColor: Colors.transparent,
```

**Comment by Abhijit-Revamp** [comment]
[✓] Flutter (Channel master, 2.4.0-1.0.pre.86, on macOS 11.4 20F71 darwin-x64, locale en-IN)
    • Flutter version 2.4.0-1.0.pre.86 at /Volumes/Abhi-Data/FlutterDev/SDK/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 0ad7b0e4cb (3 hours ago), 2021-07-06 05:06:02 -0400
    • Engine revision bd2c477177
    • Dart version 2.14.0 (build 2.14.0-281.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Volumes/Abhi-iTunes/AndroidSDK
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Volumes/Abhi-iTunes/AndroidSDK
    • ANDROID_SDK_ROOT = /Volumes/Abhi-iTunes/AndroidSDK
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk1.8.0_251.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.4 20F71 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 91.0.4472.114

**Comment by darshankawar** [comment]
Thanks for the details. I see the same behavior using latest mater version.

```
return MaterialApp(
        home: Scaffold(
        backgroundColor: Colors.transparent,
          body: Center(
            child: Container(
              color: Colors.blue,
            )
          )
    ));
```

<details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel master, 2.4.0-1.0.pre.94, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.4.0-1.0.pre.94 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ba143ae21e (2 hours ago), 2021-07-06 22:01:02 -0700
    • Engine revision 60eea42346
    • Dart version 2.14.0 (build 2.14.0-284.0.dev)

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 91.0.4472.114

• No issues found!



```

</details>

**Comment by yjbanov** [comment]
@sigmundch is it possible to remove UI injected by Dart compilers? I think instead it would be better for the compiler to provide an API and let Flutter handle the UI.

**Comment by sigmundch** [comment]
@yjbanov can you provide some additional context. What UI is being in injected?

**Comment by sigmundch** [comment]
To clarify, ddc and dart2js should not be injecting any code.

Searching for `flutter-loader`, I can find is this bootstrap codegen in flutter tools: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/web/bootstrap.dart

Is that what's injecting the extra UI in this case?

/cc @jonahwilliams

**Comment by jonahwilliams** [comment]
This should have been removed by the dart-app-ready event which was previously emitted by either ddc or dwds:

https://github.com/flutter/flutter/blob/025a3a7da312e0f6bdf77d5a1f1358bc538eb586/packages/flutter_tools/lib/src/web/bootstrap.dart#L91

**Comment by sigmundch** [comment]
indeed - the event is fired by dwds (in the client.js file that is injected onto the page).

How is this file injected by flutter tools? Is it possible that users with a custom index.html may be missing this file and the debugger is not setup properly (and as a result the event here is never fired)?

**Comment by annagrin** [comment]
the bootstrap file is generated and uploaded to chrome on first upload: https://github.com/flutter/flutter/blob/2a4e573dcdf5455ebdf0a2b3541e190021a649f1/packages/flutter_tools/lib/src/isolated/devfs_web.dart#L816

/cc @christopherfujino

**Comment by annagrin** [comment]
[dart web gardening] @christopherfujino could you please triage this one or close if it is not going to be fixed?

**Comment by yjbanov** [comment]
I just checked, and in all the following cases the animation is being removed as soon as the app loads:

- `flutter run -d chrome` - the tool launches Chrome with the indicator, then removes it when the app loads.
- `flutter run -d web-server --debug` - I checked Chrome, Safari, and Firefox, and in all cases the UI is removed after the app loads.

So I'm going to consider this particular issue as fixed. If we want to have an option to remove the debug loading UI entirely, let's file a new feature request, if it's important.
